Testscripts/Linux/nvme_blkdiscard.sh

```diff
@@ -22,7 +22,7 @@ Run_Blkdiscard() {
     update_repos
     install_package "nvme-cli"
     # Count NVME namespaces
-    namespace_count=$(echo /dev/*nvme*n[0-9] | wc -w)
+    namespace_count=$(ls -l /dev | grep -w nvme[0-9]n[0-9]$ | awk '{print $10}' | wc -l)
```

Comment:
This is a good point. The echo method is based on shellcheck recommendation as an alternative to the ls way, but in this case you're right, it will missmatch in case the pattern doesn't exist.

if you do a simple ls without the -l flag, you don't need also the awk pipe, as ls will simply output the devices matching the pattern.

Also, grep has a built-in counter with the -c flag, so the below should do the same thing:
ls /dev/ | grep -wc nvme[0-9]n[0-9]$

Please double-check this to make sure.

Reply:
Thanks Chris. Double check it.